src/sema/function_annotation.rs

```diff
@@ -1,18 +1,139 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use super::{
-    ast::{ConstructorAnnotation, Diagnostic, Namespace, Type},
+    ast::{ConstructorAnnotation, Diagnostic, Expression, Function, Namespace, Type},
     diagnostics::Diagnostics,
-    expression::{expression, ExprContext, ResolveTo},
+    expression::{expression, hex_number_literal, number_literal, ExprContext, ResolveTo},
     unused_variable::used_variable,
     Symtable,
 };
 use crate::Target;
+use num_bigint::BigInt;
+use num_traits::{One, ToPrimitive};
 use solang_parser::pt::{self, CodeLocation};
 
+/// Resolve the prototype annotation for functions (just the selector). These
+/// annotations can be resolved without a body of a function, which means they
+/// do not need to access the symbol table, like `@seed(foo)` annotations do.
+pub fn function_prototype_annotations(
+    func: &mut Function,
+    annotations: &[&pt::Annotation],
+    ns: &mut Namespace,
+) {
+    let mut diagnostics = Diagnostics::default();
+
+    for annotation in annotations {
+        match annotation.id.name.as_str() {
+            "selector" => function_selector(func, annotation, &mut diagnostics, ns),
+            _ if !func.has_body => {
+                // function_body_annotations() is not called if there is no body
```

Comment:
I think it is called when there is a body, isn't it?

Reply:
It is called when there is a body, but if there is no body then we want to generate an error, because it is not called.

Reply:
I've changed it to `// function_body_annotations() is called iff there is a body`